src/mesh_smoothing.rs

```diff
@@ -36,40 +47,111 @@ pub fn laplacian_smoothing(geometry: &Geometry, iterations: u8) -> Geometry {
 #[allow(dead_code)]
 pub fn laplacian_smoothing_with_anchors(
     geometry: &Geometry,
-    iterations: u8,
+    iterations: u32,
+    fixed_vertex_indices: &[u32],
+) -> Geometry {
+    let vertex_to_vertex_topology = mesh_topology_analysis::vertex_to_vertex_topology(geometry);
+    let (g, _, _) = laplacian_smoothing_with_anchors_full(
+        geometry,
+        vertex_to_vertex_topology,
+        iterations,
+        fixed_vertex_indices,
+        false,
+    );
+    g
+}
+
+/// Relaxes angles between mesh edges, while keeping some vertices anchored,
+/// resulting in an evenly distributed geometry stretched between the anchor
+/// points.
+///
+/// The number of vertices, faces and the overall topology remains unchanged.
+/// The more iterations, the smoother result. Too many iterations may cause slow
+/// calculation time. Only the maximum number of iterations is specified, the
+/// smoothing stops when the geometry stops transforming between iterations.
+///
+/// The algorithm is based on replacing each vertex position with an average
+/// position of its immediate neighbors
+///
+/// - `geometry` - mesh geometry to relax
+/// - `max_iterations` - maximum number of times the smoothing algorithm should
+///   relax the geometry
+/// - `fixed_vertex_indices` - indices of vertices to keep fixed during the
+///   relaxation
+#[allow(dead_code)]
+pub fn laplacian_smoothing_with_anchors_stop_when_stable(
+    geometry: &Geometry,
+    max_iterations: u32,
     fixed_vertex_indices: &[u32],
 ) -> Geometry {
-    if iterations == 0 {
-        return geometry.clone();
+    let vertex_to_vertex_topology = mesh_topology_analysis::vertex_to_vertex_topology(geometry);
+    let (g, _, _) = laplacian_smoothing_with_anchors_full(
+        geometry,
+        vertex_to_vertex_topology,
+        max_iterations,
+        fixed_vertex_indices,
+        true,
+    );
+    g
+}
+
+/// returns (smooth_geometry: Geometry, executed_iterations: u32, stable: bool)
+pub fn laplacian_smoothing_with_anchors_full(
+    geometry: &Geometry,
+    vertex_to_vertex_topology: HashMap<u32, SmallVec<[u32; 8]>>,
+    max_iterations: u32,
+    fixed_vertex_indices: &[u32],
+    stop_when_stable: bool,
+) -> (Geometry, u32, bool) {
+    if max_iterations == 0 {
+        return (geometry.clone(), 0, stop_when_stable);
```

Comment:
Consistency nit: I think `false` is more correct than `stop_when_stable` here. The changes could not have stabilized if there were no changes.